### PR TITLE
[IMP] pos, hw_drivers, hw_posbox_homepage: Six terminals 

### PR DESCRIPTION
**Pull request by aprieels**
Add support for Six payment terminals.

TaskId: 1932416

Linked to odoo/enterprise#4435

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/hw_posbox_homepage/controllers/main.py [hunk 1]
```diff
@@ -294,3 +303,28 @@ def enable_ngrok(self, auth_token):
...
+        return six_payment_terminal_template.render({
+            'title': 'Six Payment Terminal',
+            'breadcrumb': 'Six Payment Terminal',
+            'terminalId': self.get_six_terminal(),
+        })
+
+    @http.route('/six_payment_terminal_add', type='http', auth='none', cors='*', csrf=False)
+    def add_six_payment_terminal(self, terminal_id):
+        subprocess.check_call("sudo mount -o remount,rw /", shell=True)
+        with open("/home/pi/odoo-six-payment-terminal.conf", "w") as f:
```
Comment (qle-odoo) [inline addons/hw_posbox_homepage/controllers/main.py]: Be careful and do not use `shell = True` instead uses tagadad
cfr doc: https://docs.python.org/2/library/subprocess.html#frequently-used-arguments
in this case it is not problematic, but take good habits ....

File: addons/hw_posbox_homepage/controllers/main.py [hunk 2]
```diff
@@ -72,6 +73,13 @@ def get_pos_device_status(self):
             statuses[driver] = hw_proxy.drivers[driver].get_status()
         return statuses
 
+    def get_six_terminal(self):
+        path = Path.home() / 'odoo-six-payment-terminal.conf'
+        if path.exists():
+            with path.open('r') as f:
+                return f.readline().strip('\n')
```
Comment (qle-odoo) [inline addons/hw_posbox_homepage/controllers/main.py]: remove `.strip('\n')`

File: addons/hw_posbox_homepage/controllers/main.py [hunk 3]
```diff
@@ -294,3 +303,28 @@ def enable_ngrok(self, auth_token):
...
+        with open("/home/pi/odoo-six-payment-terminal.conf", "w") as f:
+            f.write(terminal_id)
+        subprocess.check_call(["sudo", "mount", "-o", "remount,ro", "/"])
+        return 'http://' + get_ip() + ':8069'
+
+    @http.route('/six_payment_terminal_clear', type='http', auth='none', cors='*', csrf=False)
+    def clear_six_payment_terminal(self):
+        subprocess.check_call(["sudo", "mount", "-o", "remount,rw", "/"])
+        path = "/home/pi/odoo-six-payment-terminal.conf"
+        if os.path.exists(path):
```
Comment (qle-odoo) [inline addons/hw_posbox_homepage/controllers/main.py]: path = Path.home() / 'odoo-six-payment-terminal.conf'

File: addons/hw_posbox_homepage/views/six_payment_terminal.html [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+        $(document).ready(function () {
+            $('#terminal-id').submit(function(e){
+                e.preventDefault();
+                $('.loading-block').removeClass('o_hide');
+                $.ajax({
+                    url: '/six_payment_terminal_add',
+                    type: 'post',
+                    data: $('#terminal-id').serialize(),
+                }).done(function (message) {
+                    location.reload(true);
```
Comment (qle-odoo) [inline addons/hw_posbox_homepage/views/six_payment_terminal.html]: message is useless
and `location.reload(false)`

**Comment by pimodoo** [comment]
@qle-odoo I think you should rebase this branch and/or remove commit 'small change for enterprise' as the file pollyfill.js is not needed anymore in this pull request: https://github.com/odoo/odoo/pull/33398

File: addons/point_of_sale/static/src/js/screens.js [hunk 5]
```diff
@@ -1926,12 +1926,7 @@ var PaymentScreenWidget = ScreenWidget.extend({
         }
 
         var lines = order.get_paymentlines();
-        var due   = order.get_due();
-        var extradue = 0;
-        if (due && lines.length  && due !== order.get_due(lines[lines.length-1])) {
-            extradue = due;
-        }
-
+        var extradue = this.compute_extradue(order);
```
Comment (pimodoo) [inline addons/point_of_sale/static/src/js/screens.js]: is it used somewhere else? an enterprise PR?

Comment (pimodoo) [reply addons/point_of_sale/static/src/js/screens.js]: sorry, I've found the enterprise one

**Comment by qle-odoo** [comment]
@robodoo r+ merge

**Pull request merged**